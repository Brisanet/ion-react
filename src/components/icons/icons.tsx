import DOMPurify from 'dompurify';
import { iconsPaths, iconType } from './svgs/icons';

export type SvgModule = typeof import('./svgs/icons');
export type IconPaths = keyof typeof iconsPaths;

export interface IonIconProps {
  type: iconType;
  size?: number;
  color?: string;
}

const iconDefault: IonIconProps = {
  type: 'access',
  color: '#282B33',
  size: 24,
};

function pathPurify(iconType: iconType): string {
  const iconPath = iconsPaths[iconType];
  const iconPathPurify = DOMPurify.sanitize(`<svg> ${iconPath} </svg>`);
  const pathWithoutSvg = iconPathPurify.substring(5, iconPathPurify.length - 6);
  return pathWithoutSvg;
}

const IonIcon = ({
  type = iconDefault.type,
  color = iconDefault.color,
  size = iconDefault.size,
}: IonIconProps) => {
  const iconPath = pathPurify(type);
  const icon = (
    <svg
      viewBox="0 0 24 24"
      height={`${size}px`}
      width={`${size}px`}
      fill={color}
      dangerouslySetInnerHTML={{
        __html: iconPath,
      }}
    />
  );

  return icon;
};

export default IonIcon;
