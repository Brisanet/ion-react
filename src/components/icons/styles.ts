import styled, { RuleSet, css } from 'styled-components';
import { convertSizeValue } from '../utils/convertSizeValue';

type IconProps = {
  $size: number | string;
  $color: string;
  $isHighlighted?: boolean;
};

const size: (size: number | string) => RuleSet<object> = (size) => {
  return css`
    width: ${convertSizeValue(size)};
    height: ${convertSizeValue(size)};
  `;
};

const color: (color: string) => RuleSet<object> = (color) => {
  return css`
    background-color: ${color};
  `;
};

export const IconHighlight = styled.div<IconProps>`
  ${({ theme, $size, $color, $isHighlighted }) => css`
    ${size($size)};
    ${color($color)}
    ${$isHighlighted
      ? css`
          ${theme.utils.flex.center()};
          border-radius: 50%;
        `
      : css`
          display: inline-flex;
        `}
  `}
`;

export const Icon = styled.svg<IconProps>`
  ${({ $size, $color }) => css`
    ${size($size)};
    fill: ${$color};
  `}
`;
