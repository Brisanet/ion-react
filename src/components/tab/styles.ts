import stitches from '../../stitches.config';
import { spacing } from '../utils/spacing';

const { styled } = stitches;

export const TabStyles = styled('button', {
  fontStyle: 'normal',
  fontWeight: '600',
  color: '$neutral7',
  background: '$neutral1',
  cursor: 'pointer',
  display: 'flex',
  alignItems: 'center',
  gap: spacing(1),

  '&:hover': {
    color: '$primary5',
    background: '$primary1',
    borderColor: '$primary3',
  },
  '&:disabled': {
    color: '$neutral5',
    background: '$neutral3',
    borderColor: '$neutral5',
    cursor: 'not-allowed',
  },
  '&:focus': {
    color: '$primaryColor',
    background: '$neutral1',
    borderColor: '$primaryColor',
    fontStyle: 'normal',
    fontWeight: '600',
    cursor: 'pointer',
    display: 'flex',
    alignItems: 'center',
  },

  variants: {
    size: {
      sm: {
        padding: '8px 16px',
        fontSize: '12px',
        lineHeight: '16px',
        gap: spacing(1),
      },
      md: {
        padding: '10px 24px',
        fontSize: '14px',
        lineHeight: '20px',
        gap: spacing(1.5),
      },
      lg: {
        padding: '12px 32px',
        fontSize: '16px',
        lineHeight: '24px',
        gap: spacing(2),
      },
    },
    direction: {
      bottom: {
        borderBottom: '2px solid $neutral3',
        borderRadius: '6px 6px 0px 0px',
      },
      top: {
        borderTop: '2px solid $neutral3',
        borderRadius: '0px 0px 6px 6px',
      },
      right: {
        borderRight: '2px solid $neutral3',
      },
      left: {
        borderLeft: '2px solid $neutral3',
      },
    },
    selected: {
      true: {
        color: '$primaryColor',
      },
    },
  },

  compoundVariants: [
    {
      selected: true,
      css: {
        background: '$neutral1',
        borderColor: '$primaryColor',
        '&:hover': {
          color: '$primary5',
          background: '$primary1',
          borderColor: '$primary3',
          '&:active': {
            color: '$primary7',
            background: '$primary2',
            borderColor: '$primary5',
          },
          '&:disabled': {
            color: '$neutral5',
            background: '$neutral3',
            borderColor: '$neutral5',
            cursor: 'not-allowed',
          },
        },
      },
    },
  ],
});
