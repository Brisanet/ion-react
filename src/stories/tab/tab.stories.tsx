import { ComponentStory, ComponentMeta } from '@storybook/react';

import IonTab, { TabProps } from '../../components/tab/tab';

export default {
  title: 'Ion/Navigation/Tab',
  component: IonTab,
} as ComponentMeta<typeof IonTab>;

const Template: ComponentStory<typeof IonTab> = (args: TabProps) => (
  <IonTab {...args} />
);

export const TabBasic = Template.bind({});
TabBasic.args = {
  label: 'Custom label',
};
export const TabSmall = Template.bind({});
TabSmall.args = {
  label: 'Small',
  size: 'sm',
};

export const TabMedium = Template.bind({});
TabMedium.args = {
  label: 'Medium',
  size: 'md',
};

export const TabLarge = Template.bind({});
TabLarge.args = {
  label: 'Large',
  size: 'lg',
};

export const TabBottom = Template.bind({});
TabBottom.args = {
  label: 'Bottom',
  direction: 'bottom',
};

export const TabTop = Template.bind({});
TabTop.args = {
  label: 'Top',
  direction: 'top',
};

export const TabRight = Template.bind({});
TabRight.args = {
  label: 'Right',
  direction: 'right',
};

export const TabLeft = Template.bind({});
TabLeft.args = {
  label: 'Left',
  direction: 'left',
};

export const Disabled = Template.bind({});
Disabled.args = {
  label: 'Disabled',
  disabled: true,
};

export const Selected = Template.bind({});
Selected.args = {
  label: 'Selected',
  selected: true,
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  label: 'Icon',
  icon: 'pencil',
};

export const WithBadge = Template.bind({});
WithBadge.args = {
  label: 'Badge',
  badge: { label: 10 },
};

export const IconAndBadge = Template.bind({});
IconAndBadge.args = {
  label: 'Badge',
  icon: 'pencil',
  badge: { label: 10 },
};
