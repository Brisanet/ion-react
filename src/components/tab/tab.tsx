import React, { useEffect, useState } from 'react';
import { SizeType } from '../../core/types/size';
import IonIcon from '../icons/icons';
import { iconType } from '../icons/svgs/icons';
import { TabStyles } from './styles';

type StitchesTabProps = React.ComponentProps<typeof TabStyles>;

type TabSizes = Exclude<SizeType, 'xs'>;

export type TabProps = {
  label: string;
  direction?: string;
  size?: TabSizes;
  selected?: boolean;
  disabled?: boolean;
  icon?: iconType;
  handleClick?: () => void;
} & StitchesTabProps;

const IonTab = ({
  label,
  icon,
  size = 'sm',
  direction = 'bottom',
  selected = false,
  disabled = false,
  handleClick,
}: TabProps) => {
  const [selectedState, setSelectedState] = useState<boolean>(selected);

  /**
   * If the @component IonTab is used individually outside
   * the @param handleClick function that changes the
   * 'selected' status will be defined to support individual IonTabs
   */
  if (handleClick === undefined) {
    handleClick = () => {
      setSelectedState(true);
    };
  }

  useEffect(() => {
    setSelectedState(selected);
  }, [selected]);

  const iconSize = {
    sm: 16,
    md: 20,
    lg: 24,
  };

  return (
    <>
      <TabStyles
        size={size}
        direction={direction}
        onClick={handleClick}
        selected={selectedState}
        disabled={disabled}
      >
        <div>
          {icon && <IonIcon type={icon} size={iconSize[`${size}`]} />}
          <span>{label}</span>
        </div>
      </TabStyles>
    </>
  );
};

export default IonTab;
