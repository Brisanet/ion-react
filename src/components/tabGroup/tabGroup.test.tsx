import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { iconType } from '../icons/svgs/icons';
import { TabProps } from '../tab/tab';
import IonTabGroup, { IonTabGroupProps, TabGroupSizes } from './tabGroup';

const mockClick = jest.fn();
const mockProps: IonTabGroupProps = {
  tabs: [
    {
      label: 'Tab 1',
      selected: true,
    },
    {
      label: 'Tab 2',
    },
    {
      label: 'Tab 3',
    },
  ],
  handleSelectedTab: mockClick,
};
const sut = (props: IonTabGroupProps = mockProps) => {
  render(<IonTabGroup {...props} />);
};

const getTabGroup = () => screen.getByTestId('ion-tabGroup');
const getTabs = () => screen.getAllByTestId('ion-tab');
const getBadge = () => screen.getAllByTestId('ion-badge');

describe('IonTabGroup', () => {
  describe('Default', () => {
    it('should render TabGroup in horizontal align by default', async () => {
      sut();
      expect(getTabGroup().className).toContain('align-horizontal');
    });

    it('should render tabs with border bottom by default', async () => {
      sut();
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).toContain('direction-bottom');
      });
    });

    it('should render TabGroup in vertical align by default', async () => {
      sut({ ...mockProps, align: 'vertical' });
      expect(getTabGroup().className).toContain('align-vertical');
    });

    it('should render the correct number of tabs', () => {
      sut();
      expect(getTabs()).toHaveLength(3);
    });

    it('should select a Tab when it is clicked', async () => {
      sut();
      const secondTab = getTabs()[1];
      await userEvent.click(secondTab);
      expect(secondTab.className).toContain('selected-true');
    });

    it('should contain first Tab selected by default', async () => {
      sut();
      const firstTab = getTabs()[0];
      expect(firstTab.className).toContain('selected-true');
    });
  });

  describe('Border', () => {
    it('should contain border right when direction is vertical', async () => {
      sut({ ...mockProps, align: 'vertical' });
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).toContain('direction-right');
      });
    });

    it('should contain border left when border and align are defined', async () => {
      sut({ ...mockProps, align: 'vertical', borderDirection: 'left' });
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).toContain('direction-left');
      });
    });

    it('should contain border top when border and align are defined', async () => {
      sut({ ...mockProps, align: 'horizontal', borderDirection: 'top' });
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).toContain('direction-top');
      });
    });

    it('should not contain border top when align is vertical', async () => {
      sut({ ...mockProps, align: 'vertical', borderDirection: 'top' });
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).not.toContain('direction-top');
      });
    });

    it('should not contain border right when align is horizontal', async () => {
      sut({ ...mockProps, align: 'horizontal', borderDirection: 'right' });
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).not.toContain('direction-right');
      });
    });
  });

  describe('Sizes', () => {
    it.each(['sm', 'md', 'lg'])(
      'should render tabs with %s size',
      async (size: string) => {
        sut({
          ...mockProps,
          size: size as TabGroupSizes,
        });

        getTabs().forEach((tab: HTMLElement) => {
          expect(tab.className).toContain(`size-${size}`);
        });
      }
    );

    it('should render tabs with sm size by default', async () => {
      sut();
      getTabs().forEach((tab: HTMLElement) => {
        expect(tab.className).toContain('size-sm');
      });
    });
  });

  describe('With icon', () => {
    const icons: iconType[] = ['pencil', 'close', 'access'];
    const tabsWithIcons: TabProps[] = [
      {
        label: 'Tab 1',
        icon: icons[0],
      },
      {
        label: 'Tab 2',
        icon: icons[1],
      },
      {
        label: 'Tab 2',
        icon: icons[2],
      },
    ];

    it('should render tabs with icons', async () => {
      sut({
        ...mockProps,
        tabs: tabsWithIcons,
      });

      icons.forEach((icon: iconType) => {
        expect(screen.getByTestId(`ion-icon-${icon}`)).toBeInTheDocument();
      });
    });
  });

  describe('With Badge', () => {
    const tabsWithIconsAndBadge: TabProps[] = [
      {
        label: 'Tab 1',
        badge: { label: 100 },
        icon: 'calendar',
      },
      {
        label: 'Tab 2',
        badge: { label: 2 },
        icon: 'calendar',
      },
      {
        label: 'Tab 3',
        badge: { label: 'expirado' },
        icon: 'access',
      },
      {
        label: 'Tab 4',
        icon: 'alert',
      },
    ];

    beforeEach(() => {
      sut({ ...mockProps, tabs: tabsWithIconsAndBadge });
    });

    it('should render a tab with badge', async () => {
      expect(getBadge()[0]).toBeTruthy();
    });

    it('should render a tab with 99+ when value bigger then 99', async () => {
      expect(getBadge()[0].textContent).toBe('99+');
    });

    it('should render a tab with value when less than 99', async () => {
      expect(getBadge()[1].textContent).toBe(
        tabsWithIconsAndBadge[1].badge?.label.toString()
      );
    });

    it('should render a tab with string badge', async () => {
      expect(getBadge()[2].textContent).toBe(
        tabsWithIconsAndBadge[2].badge?.label
      );
    });
  });
});
