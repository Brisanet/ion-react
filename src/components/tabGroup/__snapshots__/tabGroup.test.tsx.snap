// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IonTabGroup Default should render TabGroup in vertical align by default 1`] = `
.c1 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-right: 2px solid;
  border-radius: 6px 0px 0px 6px;
  border-color: #ced2db;
}

.c1:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c1 svg {
  fill: #505566;
}

.c1:hover,
.c1:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c1:hover svg,
.c1:focus-visible svg {
  fill: #146ff5;
}

.c1:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c1:active svg {
  fill: #06439d;
}

.c1:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c1:disabled svg {
  fill: #aeb2bd;
}

.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-right: 2px solid;
  border-radius: 6px 0px 0px 6px;
  border-color: #ced2db;
  color: #0858ce;
  border-color: #0858ce;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0 svg {
  fill: #0858ce;
}

<div
  class="c-gbGNpS c-gbGNpS-kwZaNe-align-vertical"
  data-testid="ion-tabGroup"
>
  <div
    data-testid="ion-tabGroup-container"
  >
    <button
      class="c0"
      data-testid="ion-tab"
    >
      <span>
        Tab 1
      </span>
    </button>
    <button
      class="c1"
      data-testid="ion-tab"
    >
      <span>
        Tab 2
      </span>
    </button>
    <button
      class="c1"
      data-testid="ion-tab"
    >
      <span>
        Tab 3
      </span>
    </button>
  </div>
</div>
`;

exports[`IonTabGroup Sizes should render tabs with lg size 1`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 12px 32px;
  font-size: 1.6rem;
  line-height: 2.4rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
  color: #0858ce;
  border-color: #0858ce;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0 svg {
  fill: #0858ce;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 1
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with lg size 2`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 12px 32px;
  font-size: 1.6rem;
  line-height: 2.4rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 2
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with lg size 3`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 12px 32px;
  font-size: 1.6rem;
  line-height: 2.4rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 3
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with md size 1`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 10px 24px;
  font-size: 1.4rem;
  line-height: 2rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
  color: #0858ce;
  border-color: #0858ce;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0 svg {
  fill: #0858ce;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 1
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with md size 2`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 10px 24px;
  font-size: 1.4rem;
  line-height: 2rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 2
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with md size 3`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 10px 24px;
  font-size: 1.4rem;
  line-height: 2rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 3
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with sm size 1`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
  color: #0858ce;
  border-color: #0858ce;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0 svg {
  fill: #0858ce;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 1
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with sm size 2`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 2
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with sm size 3`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 3
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with sm size by default 1`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
  color: #0858ce;
  border-color: #0858ce;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0 svg {
  fill: #0858ce;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 1
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with sm size by default 2`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 2
  </span>
</button>
`;

exports[`IonTabGroup Sizes should render tabs with sm size by default 3`] = `
.c0 {
  font-weight: 600;
  color: #505566;
  background: #fcfcfd;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 8px;
  padding: 8px 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  border-bottom: 2px solid;
  border-radius: 6px 6px 0px 0px;
  border-color: #ced2db;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

.c0 svg {
  fill: #505566;
}

.c0:hover,
.c0:focus-visible {
  color: #146ff5;
  background: #ebf3fe;
  border-color: #84b4fa;
}

.c0:hover svg,
.c0:focus-visible svg {
  fill: #146ff5;
}

.c0:active {
  color: #06439d;
  background: #b5d2fc;
  border-color: #146ff5;
}

.c0:active svg {
  fill: #06439d;
}

.c0:disabled {
  color: #aeb2bd;
  background: #e4e6eb;
  border-color: #aeb2bd;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

<button
  class="c0"
  data-testid="ion-tab"
>
  <span>
    Tab 3
  </span>
</button>
`;
